### ISSUE TITLE
Brunt-Vaisala frequency estimation function: possible tendency to overestimation?

### ISSUE THREAD

**Issue by mafavaron** [serv-terr/pbl_met#30]
The tests 1 to 3 made on BruntVaisala() function make quite well evident that, if temperature conditions typical of unstable conditions are used, then the function yields values about twice as large as the Brunt-Vaisala frequencies I've observed on fields (around 0.1e-1, to be compared with 0.18e-1 obtained during tests).

A test no.4 might be necessary, with temperature in the order of 0-5 °C instead of 20 °C, at say 1500m altitude, to see whether the values obtained become sensible.

Another point is: the Brunt-Vaisala frequency is not defined for unstable conditions, but the existing interface does not allow to discriminate. In ST-Me and other similar met processing software this means the user should provide externally to invalidate all values pertaining to neutral and convective conditions, e.g. based on the sign of $$H_{0}$$. But, why not to include this same parameter in the function interface, and let it do all the dirty work?

Just for reflection...

**Comment by mafavaron** [comment]
Ops... The values become invalid under negative temperatures! This is surely a bug.

**Comment by mafavaron** [comment]
After bug correction I've checked the formula. No question, it is nothing else than the definition of Brunt-Vaisala frequency, and so its values are neither "over" nor "under"-estimated.

They just are.

The temperature to use is not the dry bulb, but (according to the definition) the _virtual_ temperature. But this changes things only a little - only, check the comment in _pbl_thermo_ code and correct if needed.

I've also checked sensitivity to lapse rate, and it is not large.

As a suggestion, it is better to consider what comes from BruntVaisala() a qualitative indication more than quantitative evaluation. As such, its validity "even under unstable conditions" may be accepted, provided a comment is given in the routine text.

That said, I close the issue now.
